test/ui/RelevanceInspector/ExecutionReportITDSectionTest.ts

```diff
@@ -0,0 +1,165 @@
+          }
+        ]
+      };
+    };
+
+    const getExecutionReportSectionForITD = (refinedQueries: IRefinedQueriesFromTopClicks[]): IExecutionReportITDSection => {
+      return {
+        description: 'MLExtraction',
+        duration: 789,
+        name: EXECUTION_REPORT_SECTION.TOP_CLICKS,
+        result: {},
+        refinedQueries
+      };
+    };
+
+    const getKeywordExtractedRowData = (gridOptions): Dom => {
+      return gridOptions.rowData[0]['Keyword(s) extracted'];
+    };
+
+    const getLongQueryRecall = (gridOptions): string => {
+      return gridOptions.rowData[0]['Original large query'];
+    };
+
+    describe('when dealing with ML keywords extraction', () => {
+      it('should output an ag grid element as a container', async done => {
+        const { container } = await new ExecutionReportITDSection().build(
+          getExecutionReportSection(getExecutionReportSectionForITD([]), '')
+        );
+        expect(container.find('ag-grid-fresh')).toBeDefined();
+        done();
+      });
+
+      it('should output a fallback if the long query is empty', async done => {
+        const { container } = await new ExecutionReportITDSection().build(
+          getExecutionReportSection(getExecutionReportSectionForITD([]), '')
+        );
+        expect(container.text()).toContain('NO DATA AVAILABLE');
+        done();
+      });
+
+      it('should output a fallback if the long query is not empty but no refined query was identified by ML', async done => {
+        const { gridOptions } = await new ExecutionReportITDSection().build(
+          getExecutionReportSection(getExecutionReportSectionForITD([]), 'the long query is not empty')
+        );
+        expect(getKeywordExtractedRowData(gridOptions).text()).toContain('No keywords were extracted');
+        done();
+      });
+
+      describe('with keywords extracted', () => {
+        let execReport: IExecutionReport;
+
+        beforeEach(() => {
+          execReport = getExecutionReportSection(
+            getExecutionReportSectionForITD([{ q: 'query', score: 0.9 }, { q: 'long', score: 0.5 }]),
+            'the long query is not empty'
+          );
+        });
+
+        it('should output the keywords extracted if available', async done => {
+          const { gridOptions } = await new ExecutionReportITDSection().build(execReport);
+
+          expect(getKeywordExtractedRowData(gridOptions).text()).toContain('performed using Coveo Machine Learning');
+
+          expect(getKeywordExtractedRowData(gridOptions).text()).toContain('Keyword: query');
+          expect(getKeywordExtractedRowData(gridOptions).text()).toContain('Score: 0.9');
+
+          expect(getKeywordExtractedRowData(gridOptions).text()).toContain('Keyword: long');
+          expect(getKeywordExtractedRowData(gridOptions).text()).toContain('Score: 0.5');
+
+          done();
+        });
+
+        it('should highlights the keywords extracted if available', async done => {
```

Comment:
Small typo here. It should be either `should highlight` (without the 's') or `highlights`.